streamz/sources.py

```diff
@@ -132,35 +131,65 @@ class from_kafka(Source):
     consumer_params: dict
         Settings to set up the stream, see
         https://docs.confluent.io/current/clients/confluent-kafka-python/#configuration
+        https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
         Examples:
-        url: Connection string (host:port) by which to reach Kafka
-        group: Identity of the consumer. If multiple sources share the same
+        bootstrap.servers: Connection string(s) (host:port) by which to reach Kafka
+        group.id: Identity of the consumer. If multiple sources share the same
             group, each message will be passed to only one of them.
     poll_interval: number
         Seconds that elapse between polling Kafka for new messages
+    asynchronous: bool (True)
+        In order to start polling on a callback
+    loop: None or IOLoop
+        Loop in which to run polling as a coroutine
 
     Example
     -------
 
     >>> source = Stream.from_kafka(['mytopic'],
-    ...        dict(url='localhost:9092', group='streamz'))  # doctest: +SKIP
+    ...           {'bootstrap.servers': 'localhost:9092',
+    ...            'group.id': 'streamz'})  # doctest: +SKIP
     """
-    def __init__(self, topics, consumer_params, poll_interval=0.1):
-        import confluent_kafka as ck
-        IOLoop.current().add_callback(self.poll_kafka)
+    def __init__(self, topics, consumer_params, poll_interval=0.1,
+                 asynchronous=True, **kwargs):
         self.cpars = consumer_params
-        self.consumer = ck.Consumer(consumer_params)
-        self.consumer.subscribe(topics)
+        self.consumer = None
         self.topics = topics
-        self.sleep = poll_interval
+        self.poll_interval = poll_interval
 
-        super(from_kafka, self).__init__()
+        super(from_kafka, self).__init__(asynchronous=asynchronous, **kwargs)
+        self._connect_consumer()
+
+    def do_poll(self):
+        if self.consumer is not None:
+            msg = self.consumer.poll(0)
+            if msg and msg.value():
+                return msg.value()
 
     @gen.coroutine
     def poll_kafka(self):
         while True:
-            msg = self.consumer.poll(0)
-            if msg is None or msg.error():
-                yield gen.sleep(self.sleep)
+            val = self.do_poll()
+            if val:
+                yield self.emit(val)
```

Comment:
We tend to use `_emit` internally when we know we're asynchronous.  This avoids some logic and keeps the overhead low.